src/rgw/rgw_lua.cc

```diff
@@ -137,5 +143,146 @@ int delete_script(rgw::sal::RGWRadosStore* store, const std::string& tenant, opt
 
   return 0;
 }
+
+#ifdef WITH_RADOSGW_LUA_PACKAGES
+
+const std::string PACKAGE_LIST_OBJECT_NAME = "lua_package_allowlist";
+
+namespace bp = boost::process;
+
+int add_package(rgw::sal::RGWRadosStore* store, optional_yield y, const std::string& package_name, bool allow_compilation) {
+  // verify that luarocks can load this oackage
+  const auto p = bp::search_path("luarocks");
+  if (p.empty()) {
+    return -ECHILD;
+  }
+  bp::ipstream is;
+  const auto cmd = p.string() + " search --porcelain" + (allow_compilation ? " " : " --binary ") + package_name;
+  bp::child c(cmd,
+      bp::std_in.close(),
+      bp::std_err > bp::null,
+      bp::std_out > is);
+
+  std::string line;
+  bool package_found = false;
+  // TODO: yield on reading the output
```

Comment:
what does lack of yield imply for this initial version?

Reply:
lack of yield will only be an issue if we allow package reload.
currently, we support only package load on startup, so it does not matter ( will change comment)